core/windows/winmain.cpp

```diff
@@ -327,9 +332,11 @@ LRESULT CALLBACK WndProc2(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
 		}
 		// Handles the close message when a user clicks the quit icon of the window
 	case WM_CLOSE:
+	{
+		start_shutdown();
```

Comment:
This should be moved to `WM_QUIT` iirc. `WM_CLOSE` should just `PostQuitMessage()` for windows.